src/python/pants/jvm/resources.py

```diff
@@ -33,42 +37,21 @@ class JvmResourcesRequest(ClasspathEntryRequest):
 @rule(desc="Fetch with coursier")
 async def assemble_resources_jar(
     zip: ZipBinary,
-    union_membership: UnionMembership,
     request: JvmResourcesRequest,
 ) -> FallibleClasspathEntry:
-
+    # Request the component's direct dependency classpath, and additionally any prerequisite.
     # Filter out any dependencies that are generated by our current target so that each resource
     # only appears in a single input JAR.
     # NOTE: Generated dependencies will have the same dependencies as the current target, so we
     # don't need to inspect those dependencies.
```

Comment:
This comment should move onto the request field probably. You might also want to expand the name to something that conveys "ignore generated if generator is present" or something.